### PR TITLE
Upgrade to cypress v10

### PR DESCRIPTION
**Pull request by DiegoGomez01**
Hi, Trent Rand. I hope you're well.

These changes are to improve this project to allow the use of cypress v10, Let me know if you have any suggestions. 

## Changes Implemented

- Fix lint errors/warnings
- Upgrade libraries
- Changed the `peerDependencies` to accept cypress >=10 on the package.json
- Changed the flag `integrationFolder` to `specPattern` because that was a change on the new cypress version 10
- Separate the flag `config` and `config-file`. Now to add the configFile, you must include the flag `config-file`, `configFile` or `C` but if you want to add/replace configuration you can do that with `config` or 'c'
- To exclude a file (or list of files) you can use the flag `excludeSpecPattern` and add the files to exclude. i.e `cypress-utils run-parallel --excludeSpecPattern all.cy.js file2.cy.js file3.cy.js`

Thanks for this useful repo. Have a good day.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -44,17 +44,28 @@ var argv = yargs.scriptName('cypress-utils')
...
+    description: 'Set configuration values. Separate multiple values with commas. The values set here override any values set in your configuration file.',
     global: true,
   })
-  .option('integrationFolder', {
+  .option('specPattern', {
     alias: ['i', 'integration'],
     type: 'string',
-    description: 'Path to folder containing integration test files',
-    default: 'cypress/integration',
+    description: 'Path to folder containing e2e test files',
```
Comment (trentrand) [inline index.js:62]:
```suggestion
    description: 'A glob pattern of the test files to load.',
```

Comment (trentrand) [reply index.js:62]: Pulled this description from https://docs.cypress.io/guides/references/configuration#e2e but reworded it to remove support for Array type.

File: index.js [hunk 2]
```diff
@@ -123,7 +134,7 @@ function handleResults(error, results) {
   }
 
   // Clear command-line before printing results
-  process.stdout.write('\033c')
+  process.stdout.write('\x1Bc');
```
Comment (trentrand) [inline index.js:137]: I'm hesitant to make this change. I wasn't very familiar with either, but my search led to a warning that `\x1Bc` may not work properly in all situations.

https://stackoverflow.com/a/47503782/2020476

Were you seeing an issue with `\033c`?

File: index.js [hunk 3]
```diff
@@ -44,17 +44,28 @@ var argv = yargs.scriptName('cypress-utils')
...
     alias: ['i', 'integration'],
     type: 'string',
-    description: 'Path to folder containing integration test files',
-    default: 'cypress/integration',
+    description: 'Path to folder containing e2e test files',
+    default: 'cypress/e2e',
+    global: true,
+  })
+  .option('excludeSpecPattern', {
+    type: 'array',
```
Comment (trentrand) [inline index.js:67]:
```suggestion
    type: 'array',
    default: [],
```

File: index.js [hunk 4]
```diff
@@ -159,59 +170,37 @@ function printResultsForCommand(subjectName, subjectResults, command) {
...
 try {
-  specFiles = glob.sync(cypressConfig.integrationFolder + '/' + argv.testFiles)
+  specFiles = glob.sync(argv.specPattern + '/' + argv.testFiles)
     .filter(fileName => {
       return (
         argv.fileIdentifiers.some((fileIdentifier) => fileName.toLowerCase().includes(fileIdentifier))
       );
-    })
+    });
+  if (argv.excludeSpecPattern.length > 0) {
```
Comment (trentrand) [inline index.js:187]: this could fail if `argv.excludeSpecPattern` is undefined

File: package.json [hunk 5]
```diff
@@ -25,16 +25,16 @@
   },
   "homepage": "https://github.com/trentrand/cypress-utils#readme",
   "peerDependencies": {
-    "cypress": ">=5 <=8"
+    "cypress": ">=10"
   },
   "devDependencies": {
-    "cypress": "8.3.0"
+    "cypress": "10.7.0"
```
Comment (trentrand) [inline package.json:31]: maybe `^10.0.0` to support a wider range of users?

**Review by trentrand — COMMENTED** [review]
Thank you very much for the contribution!

It looks great, just a few minor points we can address before publishing!

File: index.js (hunk 1 above)
Comment (DiegoGomez01) [reply index.js:62]: Thanks for the suggestion Trent Rand. I'll do the change. It's better.

File: index.js (hunk 2 above)
Comment (DiegoGomez01) [reply index.js:137]: It works fine. But I had an eslint warning. But you're right, maybe it does not work correctly in all situations. I'll revert the change

File: index.js (hunk 3 above)
Comment (DiegoGomez01) [reply index.js:67]: Yes, that change it's important. Many thanks

File: package.json (hunk 5 above)
Comment (DiegoGomez01) [reply package.json:31]: Yes, sure.

File: index.js (hunk 4 above)
Comment (DiegoGomez01) [reply index.js:187]: I'm not sure because we're including a default value `[]` in the excludeSpecPattern. May I include the validation anyway?

**Review by trentrand — APPROVED** [review]
Well done! 👏🏻 I appreciate you contributing these changes.

I'll merge and publish version 3.0.0 of the package.

File: index.js (hunk 4 above)
Comment (trentrand) [reply index.js:188]: You're right, the code change in [58d39db](https://github.com/trentrand/cypress-utils/pull/25/commits/58d39db353ae1decd42a55df87927648b18d2b67) fixes this

**Comment by trentrand** [comment]
I tested this against the [cypress-kitchen-sink](https://github.com/cypress-io/cypress-example-kitchensink) test files.

`cypress-utils run-parallel 1-getting-started/todo 2-advanced-examples/waiting viewport window`
> image

**Comment by trentrand** [comment]
@DiegoGomez01, I amended commits 60f2f11 and ad50f08

**Comment by trentrand** [comment]
These changes were published as version 3.0.0! 🎉 

See the release for more details:
https://github.com/trentrand/cypress-utils/releases/tag/v3.0.0

**Pull request merged**